module/zfs/spa_misc.c

```diff
@@ -1798,8 +1798,24 @@ spa_get_worst_case_asize(spa_t *spa, uint64_t lsize)
 uint64_t
 spa_get_slop_space(spa_t *spa)
 {
-	uint64_t space = spa_get_dspace(spa);
-	uint64_t slop = MIN(space >> spa_slop_shift, spa_max_slop);
+	uint64_t space = 0;
+	uint64_t slop = 0;
+
+	/*
+	 * Sometimes, this is unset when we get here.
+	 * If that's the case, the adjustment below won't do
+	 * anything reasonable, so we regenerate it.
```

Comment:
How about changing this to: Make sure spa_dedup_dspace has been set

Reply:
Done.